src/components/DateTimePicker/DateTimePicker.js

```diff
@@ -0,0 +1,63 @@
+import React from "react";
+import Picker from "react-datepicker";
+import "react-datepicker/dist/react-datepicker.css";
+
+const DateTimePicker = props => {
+  const { date, setDate, customInput, customInputProps, isTimeOnly, isSmallScreen, ...rest } = props;
+
+  const fixOnEnter = e => {
+    // makes sure input loses focus on enter, to avoid date parsing bugs
+    if (e.key.toLowerCase() === "enter") {
+      setTimeout(() => document.activeElement.blur());
+    }
+  };
+
+  const handleTimeChange = newTime => {
+    if (!newTime) return;
+    const dateWithOldTime = date || new Date();
+    const hours = newTime.getHours();
+    const minutes = newTime.getMinutes();
+    const dateWithUpdatedTime = new Date(
+      dateWithOldTime.setHours(hours, minutes)
```

Comment:
It seem there is a small inconsistency in the behavior of the component here.

Following line assumes `setDate(dateWithUpdatedTime);` the DateTimePicker component does not mutate the Date object that was provided via date property, but rather invokes setDate callback with a new date object and let parent component decide what needs to be done. If that assumption is correct, then we probably should avoid avoiding mutating date property in this line `dateWithOldTime.setHours(hours, minutes)`.

Reply:
You are right, now i create a new Date object from date and work with that instead